osquery/tables/system/windows/cpu_time.cpp

```diff
@@ -0,0 +1,80 @@
+/**
+ * Copyright (c) 2014-present, The osquery authors
+ *
+ * This source code is licensed as defined by the LICENSE file found in the
+ * root directory of this source tree.
+ *
+ * SPDX-License-Identifier: (Apache-2.0 OR GPL-2.0-only)
+ */
+
+#include "osquery/core/windows/wmi.h"
+#include <osquery/core/system.h>
+#include <osquery/core/tables.h>
+#include <osquery/logger/logger.h>
+#include <osquery/sql/sql.h>
+#include <osquery/utils/conversions/tryto.h>
+
+#include <unordered_map>
+
+namespace osquery {
+namespace tables {
+
+QueryData genCpuTime(QueryContext& context) {
+  QueryData results;
+
+  const Expected<WmiRequest, WmiError> wmiSystemReq_uptime =
+      WmiRequest::CreateWmiRequest(
+          "select ElapsedTime, Name from "
+          "Win32_PerfFormattedData_PerfProc_Process");
+  if (!wmiSystemReq_uptime || wmiSystemReq_uptime->results().empty()) {
+    LOG(WARNING) << "Error retrieving information from WMI.";
+    return results;
+  }
+
+  const std::vector<WmiResultItem>& uptimeData = wmiSystemReq_uptime->results();
+  std::unordered_map<std::string, long long> uptimeMap;
+  for (const auto& data : uptimeData) {
+    std::string name;
+    data.GetString("Name", name);
+    long long temp = 0;
+    data.GetLongLong("ElapsedTime", temp);
+    uptimeMap[name] = temp; // in seconds
+  }
+
+  const Expected<WmiRequest, WmiError> wmiSystemReq =
+      WmiRequest::CreateWmiRequest(
+          "select Name, PercentUserTime, PercentPrivilegedTime, "
+          "PercentIdleTime, PercentInterruptTime, PercentPriorityTime "
+          "from Win32_PerfFormattedData_Counters_ProcessorInformation");
+  if (!wmiSystemReq || wmiSystemReq->results().empty()) {
+    LOG(WARNING) << "Error retrieving information from WMI.";
+    return results;
+  }
+
+  const std::vector<WmiResultItem>& wmiResults = wmiSystemReq->results();
+  for (const auto& data : wmiResults) {
+    std::string name;
+    data.GetString("Name", name);
+    r["core"] = name;
+    long long uptime = uptimeMap[name];
+    long percent = 0;
+    data.GetLongLong("PercentUserTime", percent);
+    // Hundredths of a second, percent / 100 * uptime * 100
+    r["user"] = BIGINT(percent * uptime);
```

Comment:
Hrm. I think this is going to be weird. We already see some issues around `uptime` not behaving as expected, and I think the lack of precision in percent is going to make this jump around.

Reply:
Thanks for the help! I've moved the spec file.
---
I'm still looking for a better source of uptime or specific times in windows, and yeah, I think the precision is a real problem - elapsed time given by wmi is in seconds as int, and the percentages are also integers. (It probably didn't make sense to use 0.01 sec as the unit.)

In case it doesn't work out, do you think it's a good idea to provide the percents instead, maybe in a new table just for windows?

Reply:
If you move or add a spec file (any file that contributes in the build really), you also have to modify the respective CMakeLists.txt.
In this case https://github.com/osquery/osquery/blob/master/specs/CMakeLists.txt; you would have to move the file from the platform_dependent to the independent one.

Reply:
Thanks for the guidance!